File: csharp/RoseGardenTest.cs
using NUnit.Framework;
using System.Collections.Generic;

namespace csharp
{
    [TestFixture]
    public class RoseGardenTest
    {
        [Test]
        public void foo()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            RoseGarden app = new RoseGarden(Items);
            app.UpdateQuality();
            Assert.AreEqual("fixme", Items[0].Name);
        }
    }
}

File: Java/src/test/java/com/rosegarden/RoseGardenTest.java
package com.rosegarden;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class RoseGardenTest {

    @Test
    void foo() {
        Item[] items = new Item[] { new Item("foo", 0, 0) };
        RoseGarden app = new RoseGarden(items);
        app.updateQuality();
        assertEquals("fixme", app.items[0].name);
    }

}
